Framework/DataHandling/src/LoadSpiceXML2DDet.cpp

```diff
@@ -460,6 +476,80 @@ LoadSpiceXML2DDet::parseSpiceXML(const std::string &xmlfilename) {
   return vecspicenode;
 }
 
+//----------------------------------------------------------------------------------------------
+/// parse binary integer file
+std::vector<unsigned int>
+LoadSpiceXML2DDet::binaryParseIntegers(std::string &binary_file_name) {
+  // check binary file size
+  ifstream infile(binary_file_name.c_str(), ios::binary);
+  streampos begin, end;
+  begin = infile.tellg();
+  infile.seekg(0, ios::end);
+  end = infile.tellg();
+  g_log.information() << "File size is: " << (end - begin) << " bytes.\n";
+
+  size_t num_unsigned_int =
+      static_cast<size_t>(end - begin) / sizeof(unsigned int);
+  size_t num_dets = num_unsigned_int - 2;
+  g_log.information() << "File contains " << num_unsigned_int
```

Comment:
You should probably check that `num_unsigned_int` is `>=2` before you try to read the first 2 values. If I create an empty file (`touch empty_file.bin`) and try to read it I get
```
LoadSpiceXML2DDet-[Information] File size is: 0 bytes.
LoadSpiceXML2DDet-[Information] File contains 0 unsigned integers and thus 18446744073709551614 detectors.
LoadSpiceXML2DDet-[Error] Error in execution of algorithm LoadSpiceXML2DDet:
LoadSpiceXML2DDet-[Error] std::bad_alloc
```

Reply:
Applied.